packages/usa-header/src/index.js

```diff
@@ -188,57 +189,83 @@ const handleEscape = (event) => {
   focusNavButton(event);
 };
 
+/**
+ * Takes current url and the url from the clicked link, then compares link origin and path without any hash to
+ * deterimine if the link is to a subsection of current page.
+ *
+ * @param {HTMLAnchorElement} targetLink - Target link clicked in header.
+ * @returns {boolean} Return `true` if target is same page || Return `false` if target is on a different page.
+ */
+const isChildSection = (targetLink) => {
+  const baseURL = new URL(CURRENT_PAGE);
+  const linkDestination = new URL(targetLink.href);
+  const currentURLPath = baseURL.origin + baseURL.pathname;
+  const linkURLPath = linkDestination.origin + linkDestination.pathname;
+
+  return currentURLPath === linkURLPath;
+};
```

Comment:
## Improve is child section logic

RE: @mejiaj comment:

>Can you share an example use case?
>
>If we need to check that the links href is part of the current page there are a few ways to do this. For example, you could also use includes() to compare strings.
>
>Pseudocode example
 ```js
const isChildSection = (targetLink) => (window.location.href).includes(targetLink.href);
```
> If you only need to read the values you can do baseURL.pathname and targetLink.pathname.

I found that using `include` causes unintentional behavior when the link path is deeper but contains the previous path.

Take the patterns page on our site for example:

```
designsystem.digital.gov/ → /patterns/ → /patterns/intro
```

Since each page includes the previous path, each time we click into a nested page from the mobile nav, it will close despite it loading a new page. 

Instead we should check that the base url and path before `#` match exactly to indicate it is in fact the same page.

I improved the check to include the link origin (base url).

Let me know if you have any other questions or concerns! 👍